### PR TITLE
Add write004 for prometheus CollectorRegistry

### PR DESCRIPTION
**Pull request by iodkowski**
Source: https://github.com/ChristopherDavenport/epimetheus/blob/master/core/src/main/scala/io/chrisdavenport/epimetheus/CollectorRegistry.scala#L46

### CODE REVIEW

**Comment by t3hnar** [comment]
@iodkowski could you please explain what you wanted to achieve with these changes?

**Comment by iodkowski** [comment]
> @iodkowski could you please explain what you wanted to achieve with these changes?

Just to have opportunity write metrics to String (for example it can be used for endpoints)

**Comment by t3hnar** [comment]
@iodkowski but why not just original `CollectorRegistry` from prometheus directly without adding that to interface which abstract over the prometheus ?

File: modules/prometheus/src/main/scala/com/evolutiongaming/smetrics/Prometheus.scala [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        magnet.withValues { labelValues =>
+          if (labelValues.isEmpty) collector.as[R]
+          else collector.labels(labelValues: _*).as[R]
+        }
+      }
+    }
+
+    new Prometheus[F] {
+
+      val registry: CollectorRegistry[F] = new CollectorRegistry[F] {
```
Comment (t3hnar) [inline modules/prometheus/src/main/scala/com/evolutiongaming/smetrics/Prometheus.scala]: I think that we can leave CollectorRegistryPrometheus as is and just use it here, so:

the diff is smaller
history is still in place
this way it is the most flexible
you won't even break backward compatibli

File: modules/prometheus/src/main/scala/com/evolutiongaming/smetrics/Prometheus.scala [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+import cats.effect.{Resource, Sync}
+import cats.implicits._
+import io.prometheus.client.Collector
+import io.prometheus.client.exporter.common.TextFormat
+import io.prometheus.{client => P}
+
+trait Prometheus[F[_]] {
+
+  val registry: CollectorRegistry[F]
```
Comment (t3hnar) [inline modules/prometheus/src/main/scala/com/evolutiongaming/smetrics/Prometheus.scala]: please define these always as `def` in basic interface, however implement via `val` in the implementation

File: modules/prometheus/src/main/scala/com/evolutiongaming/smetrics/Prometheus.scala [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+  val write004: F[String]
```
Comment (t3hnar) [inline modules/prometheus/src/main/scala/com/evolutiongaming/smetrics/Prometheus.scala]: `def` here as well

**Comment by x0ji0d** [comment]
Oh my, I can't manage github conversations :(
@t3hnar should we also add `val default: F[Prometheus]`?

**Comment by t3hnar** [comment]
> Oh my, I can't manage github conversations :(
> @t3hnar should we also add `def default[F[_]]: Prometheus[F]`?

Yes, I think that's a nice thing todo

**Pull request merged**